### PR TITLE
fix(companion): edge-trigger native window dragging

### LINKED ISSUE #1219: [Bug]: Companion native drag can leave pointer stuck on KDE Wayland
### What happened, and what did you expect?

On KDE Plasma Wayland, clicking and dragging the Desktop Companion can leave the pointer stuck in a native window-drag state even after the physical mouse button has been released.

Once this happens, the pointer remains captured by the Companion and normal mouse movement or clicks do not release it. In my case, the only reliable way to recover is to hold `Escape` while quickly moving the pointer outside the Companion window.

Expected behavior: releasing the mouse button should immediately end the native window drag and release the pointer. Subsequent left-click and right-click interactions should continue to work normally.

I traced the issue to the Companion repeatedly sending `ViewportCommand::StartDrag` on every repaint while `primary_down()` remains true. Changing it to trigger `StartDrag` only on the `primary_pressed()` edge resolved the issue in repeated testing on KDE Wayland.

### Steps to reproduce

1. Run the Desktop Companion on KDE Plasma under Wayland.
2. Left-click and drag the Companion.
3. Release the physical mouse button.
4. The pointer may remain stuck on the Companion as if the mouse button were still held.
5. Try moving the pointer away or using left/right click again; the drag state remains stuck.

In my case, recovery requires holding `Escape` while quickly moving the pointer outside the Companion window.

### oh-my-opencode.json

```jsonc
{
  "$schema": "https://unpkg.com/oh-my-opencode-slim@latest/oh-my-opencode-slim.schema.json",
  "preset": "free",
  "showStartupToast": false,
  "companion": {
    "enabled": true,
    "position": "bottom-right",
    "size": "medium",
    "gifPack": "default",
    "loopStyle": "classic",
    "speed": 1,
    "debug": false
  },
  "presets": {
    "openai": {
      "orchestrator": {
        "model": "openai/gpt-5.6-terra",
        "variant": "high",
        "skills": [
          "*",
          "!make-interfaces-feel-better"
        ],
        "mcps": []
      },
      "oracle": {
        "model": "openai/gpt-5.6-sol",
        "variant": "high",
        "skills": [],
        "mcps": [
          "codebase-memory-mcp"
        ]
      },
      "librarian": {
        "model": "openai/gpt-5.6-luna",
        "variant": "low",
        "skills": [],
        "mcps": [
          "context7",
          "github",
          "tavily",
          "gh_grep"
        ]
      },
      "explorer": {
        "model": "openai/gpt-5.6-luna",
        "variant": "low",
        "skills": [],
        "mcps": []
      },
      "designer": {
        "model": "openai/gpt-5.6-luna",
        "variant": "medium",
        "skills": [],
        "mcps": [
          "chrome-devtools"
        ]
      },
      "fixer": {
        "model": "openai/gpt-5.6-luna",
        "variant": "xhigh",
        "skills": [],
        "mcps": [
          "chrome-devtools"
        ]
      },
      "graph-auditor": {
        "model": "openai/gpt-5.6-terra",
        "variant": "high",
        "orchestratorPrompt": "@graph-auditor\n- Role: Read-only bounded Tier 3 graph auditor with explicit coverage verification and source fallback.\n- Delegate when: a bounded audit requires evidence of completeness, exhaustive caller/impact analysis, dead-code or absence claims, or finding every affected path.\n- Do not delegate for: ordinary code search, routine debugging, architecture advice, code review, or implementation; use explorer, oracle, or fixer instead.",
        "skills": [],
        "mcps": [
          "codebase-memory-mcp"
        ]
      }
    },
    "opencode-free": {
      "orchestrator": {
        "model": "opencode/muse-spark-1.2-contributor-free",
        "variant": "high",
        "skills": [
          "*",
          "!make-interfaces-feel-better",
          "!better-icons",
          "!ce-brainstorm",
          "!pnpm",
          "!vite",
          "!vitest-testing"
        ],
        "mcps": []
      },
      "oracle": {
        "model": "opencode/nemotron-3-ultra-free",
        "skills": [
          "ce-brainstorm"
        ],
        "mcps": [
          "codebase-memory-mcp"
        ]
      },
      "librarian": {
        "model": "opencode/ling-3.0-flash-fin-free",
        "variant": "medium",
        "skills": [],
        "mcps": [
          "context7",
          "github",
          "tavily",
          "gh_grep"
        ]
      },
      "explorer": {
        "model": "opencode/nemotron-3.5-lightning-free",
        "skills": [],
        "mcps": []
      },
      "designer": {
        "model": "opencode/mimo-v2.5-free",
        "skills": [
          "make-interfaces-feel-better",
          "better-icons"
        ],
        "mcps": [
          "chrome-devtools"
        ]
      },
      "fixer": {
        "model": "opencode/muse-spark-1.2-contributor-free",
        "variant": "xhigh",
        "skills": [
          "pnpm",
          "vite",
          "vitest-testing"
        ],
        "mcps": [
          "chrome-devtools"
        ]
      },
      "graph-auditor": {
        "model": "opencode/nemotron-3-ultra-free",
        "skills": [],
        "mcps": [
          "codebase-memory-mcp"
        ],
        "orchestratorPrompt": "@graph-auditor\n- Role: Read-only bounded Tier 3 graph auditor with explicit coverage verification and source fallback.\n- Delegate when: a bounded audit requires evidence of completeness, exhaustive caller/impact analysis, dead-code or absence claims, or finding every affected path.\n- Do not delegate for: ordinary code search, routine debugging, architecture advice, code review, or implementation; use explorer, oracle, or fixer instead."
      }
    }
  }
}
```

### OpenCode version

1.18.31

### oh-my-opencode-slim version

2.2.21

### Operating system

Manjaro Linux x86_64 - KDE Wayland

### Logs, screenshots, or extra context

Related upstream issue: https://github.com/emilk/egui/issues/7959

The Companion currently sends `ViewportCommand::StartDrag` continuously while `primary_down()` remains true. On Wayland, the pointer-down state may remain stale after native window dragging, causing `StartDrag` to be requested repeatedly.

I tested a minimal change that triggers `StartDrag` only on the `primary_pressed()` edge and removes the `primary_down()` loop. This resolved the stuck-pointer behavior in repeated testing on KDE Wayland.

Validation of the fix:

* `cargo test`: 49 passed, 0 failed
* `cargo build`: passed
* `cargo fmt --check`: passed
* `git diff --check`: passed

This appears separate from #680, which concerns click-through behavior rather than the Companion's native drag interaction.

https://github.com/user-attachments/assets/6463bb65-9afe-4e09-baae-ec2d3dbbd03f

### PR DESCRIPTION
**Pull request by Amm1rr**
## What changed

* Trigger native Companion window dragging only on the `primary_pressed()` edge.
* Stop reissuing `ViewportCommand::StartDrag` on every frame while `primary_down()` remains true.
* Add focused regression tests for the drag-start decision, ensuring a press edge can trigger at most one drag request.

## Why

On KDE Wayland, repeatedly issuing `StartDrag` could leave the Companion stuck in a native drag state even after the physical mouse button was released.

Triggering the native drag only on the initial press edge avoids repeatedly restarting the operation while preserving the existing drag and position-persistence behavior.

Manually verified on Manjaro KDE Wayland.

### Validation

* `cargo test`: 49 passed
* `cargo build`: passed
* `cargo fmt --check`: passed
* `git diff --check`: passed
* `bun run typecheck`: passed

Fixes #1219

### CODE REVIEW

File: companion/src/app.rs [hunk 1]
```diff
@@ -1183,4 +1185,42 @@ mod tests {
...
+            (false, false),
+            (false, true),
+            (false, false),
+            (false, false),
+            (false, false),
+        ];
+        let emits: Vec<bool> = frames
+            .iter()
+            .map(|(m, p)| should_start_drag(*m, *p))
+            .collect();
```
[the remark this answers is not in the document]
Comment (Amm1rr) [reply companion/src/app.rs]: Addressed in `fd9ae698`.

The drag-start path is now extracted into `handle_drag_start`, and the regression test exercises it through a real `egui::Context`, including press, held, and release frames.

The test inspects `FullOutput.viewport_output` and verifies the actual `ViewportCommand::StartDrag` emission, ensuring held frames do not re-emit the command.
